### PR TITLE
Massive Refactor

### PR DESCRIPTION
**Pull request by TomBerriot**
Add TypeScript
Removes all commands apart from dev memes
Convert to classes most files
Remove useless dependencies

### CODE REVIEW

**Review by nclsHart — CHANGES_REQUESTED** [review]
I think you should use spaces instead of tabs.

File: src/DiscordBot/Commands/helpCommand.ts [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+		this.aliases= ['h'];
+		this.usage= '[command name]';
+		this.description= 'Get explanations on the commands of the bot and the state of the server';
+	}
+
+	public async execute(message, args): Promise<any>{
+		return message.channel.send(
+			`${'```Markdown' +
+			'\n# Actual Prefix : '}${ DiscordBot.getPrefix() }\n` +
+			'\n# Commands : ' +
```
Comment (nclsHart) [inline src/DiscordBot/Commands/helpCommand.ts]: Commands:

File: src/DiscordBot/Commands/helpCommand.ts [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+		this.name= 'help';
+		this.aliases= ['h'];
+		this.usage= '[command name]';
+		this.description= 'Get explanations on the commands of the bot and the state of the server';
+	}
+
+	public async execute(message, args): Promise<any>{
+		return message.channel.send(
+			`${'```Markdown' +
+			'\n# Actual Prefix : '}${ DiscordBot.getPrefix() }\n` +
```
Comment (nclsHart) [inline src/DiscordBot/Commands/helpCommand.ts]: Actual Prefix:

File: src/DiscordBot/Commands/helpCommand.ts [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+			'\n# Commands : ' +
+			'\n   - devMemes (shortcut : d) : random dev memes across the web ' +
+			'\n   - help' +
+			'\n' +
+			'\n# GitHub :' +
+			'\n   - Feel free to collaborate on the developpement of the bot : https://github.com/Cleverdawn/pedo-dev-bot.git ' +
```
Comment (nclsHart) [inline src/DiscordBot/Commands/helpCommand.ts]: Seems outdated, use `https://github.com/TomBerriot/dev-bot` instead.

File: src/DiscordBot/Commands/helpCommand.ts [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+
+	public async execute(message, args): Promise<any>{
+		return message.channel.send(
+			`${'```Markdown' +
+			'\n# Actual Prefix : '}${ DiscordBot.getPrefix() }\n` +
+			'\n# Commands : ' +
+			'\n   - devMemes (shortcut : d) : random dev memes across the web ' +
+			'\n   - help' +
+			'\n' +
+			'\n# GitHub :' +
```
Comment (nclsHart) [inline src/DiscordBot/Commands/helpCommand.ts]: GitHub:

**Review by nclsHart — CHANGES_REQUESTED** [review]

**Comment by TomBerriot** [comment]
https://youtu.be/R-3rYIsEKYw?t=530

**Comment by TomBerriot** [comment]
👌

**Pull request closed without merging**